lib/SILGen/SILGenRequests.cpp

```diff
@@ -46,6 +46,16 @@ SourceLoc swift::extractNearestSourceLoc(const SILGenDescriptor &desc) {
   return SourceLoc();
 }
 
+evaluator::DependencySource
+SILGenSourceFileRequest::readDependencySource(Evaluator &e) const {
+  auto &desc = std::get<0>(getStorage());
+  auto *unit = desc.context.dyn_cast<FileUnit *>();
+  return {
+    dyn_cast_or_null<SourceFile>(unit),
+    evaluator::DependencyScope::Cascading
```

Comment:
We should similarly flip over to private dependencies when we SILGen or IRGen a function body.

Reply:
Actually, I wonder if all dependencies recorded during SILGen and IRGen can be non-cascading. What do you think?

Reply:
That sounds like a fine idea, but we need to get more tests up for dependencies discovered during those later phases. In fact, we need to get *any* tests for this since we currently emit reference dependencies after Sema.